doc/user/content/ops/speedup.md

```diff
@@ -0,0 +1,149 @@
+ %0 =                                                            +
+ | ReadExistingIndex materialize.public.contacts_first_name_idx  +
+ | Get materialize.public.contacts (u23)                         +
+ | Filter (#0 = "Jann")                                          +
+```
+
+`ReadExistingIndex` indicates that the query is using the `contacts_first_name_idx`. An absence of it would mean
+that the query is scanning the whole table resulting in slower results. -->
+
+## WHERE
+
+The filtering clause is one of the most used operations in any database. Set up an index over the columns that a query filters by to increase the speed.
+
+### Literal Values
+
+Filtering by literal values is a typical case. An index over the column filtered will do the work.
+
+Back to our example, to improve retrieving all the contacts by a particular first name (the literal value):
+
+```sql
+CREATE INDEX contacts_first_name_idx ON contacts (first_name);
+
+SELECT * FROM contacts WHERE first_name = 'Charlie';
+```
+
+### Expressions
+
+Expressions can also be part of the index. Materialize will calculate them faster using an index instead of calculating them on the fly for every query.
+
+E.g., Since contact names are probably not always correctly written, using a function to upper case the name is helpful.
+
+```sql
+CREATE INDEX contacts_upper_first_name_idx ON contacts (upper(first_name));
+
+SELECT * FROM contacts WHERE upper(first_name) = 'CHARLIE';
+```
+
+An _expression_ goes beyond a function. It could be a mathematical expression that is present in the filtering
+
+```sql
+CREATE INDEX contacts_upper_first_name_idx ON contacts (prefix - 1 = 0);
+
+SELECT * FROM contacts WHERE prefix - 1 = 0;
+```
+
+
+### Literal Values and Expressions
+
+It's possible to also combine both worlds, literal values and expressions.
+
+```sql
+CREATE INDEX contacts_upper_first_name_phone_idx ON contacts (upper(first_name), phone);
+
+SELECT * FROM contacts WHERE first_name = 'CHARLIE' AND phone = 873090;
+```
+
+### Compound index
+
+When using a compound index, consider how the optimizer works. It's rule-based, and if it detects an index it is helpful for a query, it will use it. Creating an index over all the columns doesn't mean that all
+all the queries will be faster. Only the ones that use all the columns will receive an improvement.
+
+E.g., There is indecision about what to index by, and a we decide to create a compound index.
+
+```sql
+CREATE INDEX contacts_all_index ON contacts(first_name, last_name, phone, prefix);
+
+-- NO improvement over this query:
+SELECT * FROM contacts WHERE first_name = 'CHARLIE' AND phone = 873090;
+
+-- Improvement over this query:
+SELECT * FROM contacts WHERE first_name = 'CHARLIE' AND last_name = 'EILR' AND phone = 873090 AND prefix = 1;
+```
+
+It is important to clarify that it is a good practice to create a compound index over all the columns of a view to distribute records between workers properly.
```

Comment:
I'm not convinced this sentence is going to help users determine which columns to put into their index. Should we just tell people to do `CREATE DEFAULT INDEX ON blah` because `DEFAULT INDEX` always does this?

Reply:
In that case I think this advice is at odds with the rest of the document. If indexes need to be over the `WHERE` clause matching exact expressions, then creating a `DEFAULT INDEX` might always miss that? Should our advice be: if you have a `WHERE` clause that's an equality filter, make your index over exactly that expression.